### PR TITLE
Add template tsconfig and makefile for scanners.

### PR DESCRIPTION
**Pull request by dj2**
# Summary | Résumé

This PR pulls the common makefile and tsconfig information up a level
and updates the two scanners to share the configuration.

### CODE REVIEW

**Comment by dj2** [comment]
@mohamed-cds this adds owasp-zap into the RESOURCES so all the make commands run against it. Was it left out on purpose? Seemed like an oversight?

**Comment by mohamed-cds** [comment]
> @mohamed-cds this adds owasp-zap into the RESOURCES so all the make commands run against it. Was it left out on purpose? Seemed like an oversight?

wasn't intentional 👍 I think it was accidently dropped during my refactor

File: .github/workflows/scripts/build_container.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
```
Comment (patheard) [inline .github/workflows/scripts/build_container.sh:2]: Might be worth enabling [bash strict mode](http://redsymbol.net/articles/unofficial-bash-strict-mode/).  It has saved me more than once:
```sh
#!/bin/bash
set -euo pipefail
IFS=$'\n\t'
```

Comment (dj2) [reply .github/workflows/scripts/build_container.sh:4]: Cool. Done.

File: .github/workflows/scripts/build_container.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+IMAGE=$3
+
+EXTRA_ARGS=()
+if [[ "$IMAGE" == "scanners/axe-core" ]];
+then
+  EXTRA_ARGS+=(--build-arg CHROMIUM=with-chromium)
+fi
+
+DOCKER_FILE="Dockerfile"
+if [[ $IMAGE == scanners/* ]];
```
Comment (patheard) [inline .github/workflows/scripts/build_container.sh:16]: nit - quotes around var and string:
```sh
if [[ "$IMAGE" == "scanners/*" ]];
```

**Review by patheard — COMMENTED** [review]
Minor quote comment, but for the good idea vs. bad idea debate, I'm all for things that get rid of duplication.

File: .github/workflows/scripts/build_container.sh (hunk 2 above)
Comment (dj2) [reply .github/workflows/scripts/build_container.sh:16]: The lack of quotes on scanners/* was intentional as I want it to glob, not strictly match.

Comment (patheard) [reply .github/workflows/scripts/build_container.sh:16]: Ah, my bad

**Review by patheard — APPROVED** [review]

**Pull request merged**